### PR TITLE
community logo: set restriction and force fresh image

### PR DESCRIPTION
**Pull request by NRodriguezcuellar**
now will set the restriction param based on the community visibility and it will force the community logo in the settings page to always update so that any changes are immediately shown to the user, no matter the browser cache.

Now also shows a message telling the user that it may take a few moments before the changes to the community logo can be seen everywhere (because of the browser caching it for a certain amount of time)

some context on this PR can be found here: https://codimd.web.cern.ch/DpHiJGHSRlGGyPg1KNTcmw#

Screenshot from 2022-07-07 13-44-18
Screenshot from 2022-07-07 14-16-21

closes https://github.com/inveniosoftware/invenio-communities/issues/718

### CODE REVIEW

File: invenio_communities/communities/resources/resource.py [hunk 1]
```diff
@@ -160,14 +160,31 @@ def rename(self):
...
-        return item.send_file(), 200
+
+        max_age = 300
+        restricted = False
+
+        if no_cache:
+            restricted = True
+
+        return (
+            item.send_file(restricted=restricted, max_age=max_age, no_cache=no_cache),
```
Comment (ntarocco) [inline invenio_communities/communities/resources/resource.py:185]: I am not sure that `max-age` and `no-cache` are needed here.

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js [hunk 2]
```diff
@@ -145,7 +145,7 @@ const LogoUploader = ({ community, defaultLogo, hasLogo, onError }) => {
             <input {...getInputProps()} />
             <Header className="mt-0">{i18next.t("Profile picture")}</Header>
             <Image
-              src={community.links.logo}
+              src={community.links.logo + "?no-cache=true"}
```
Comment (ntarocco) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:148]: Random number or milliseconds instead?

**Comment by ntarocco** [comment]
I think the code should be more self-explanatory: it is not clear, by simply reading it, why sometimes the cache is needed and some other times it is not.

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js (hunk 2 above)
Comment (NRodriguezcuellar) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:148]: it felt bad to do that :cry:

**Comment by alejandromumo** [comment]
In the screenshots the logo quota does not show any size, can you please validate wheter that is a bug? @NRodriguezcuellar

**Review by ntarocco — APPROVED** [review]

**Review by Pineirin — APPROVED** [review]

**Review by kpsherva — APPROVED** [review]

**Pull request merged**